### PR TITLE
[WIP] Integrate BibSonomy Plugin

### PR DESCRIPTION
**Pull request by zesaro**
**We try to offer the latest build at http://builds.jabref.org/bibsonomy/**. Feel free to comment on the functionality in this PR.

[BibSonomy](https://www.bibsonomy.org/) is a tool for maintaining references. [PUMA](https://puma.ub.uni-stuttgart.de/) is the instance of the University of Stuttgart. For more information, check the related work of [CloudRef](https://dblp.org/rec/html/conf/zeus/KoppBM18).

Example usage of BibSonomy:

grafik

> Aim: Get first working version into JabRef again, do some marketing, and attract a maintainer for that functionality.

The aim of this pull request is to integrate the BibSonomy Plugin from JabRef 2.11 into the actual JabRef master branch. You can run BibSonomy from the menubar (between `Tools` and `Options`), search for entries and fetch them. There are still parts in the code which has to be updated. Since 2.11 much has changed. For example the MetaData which I have to investigate.

I am looking forward to hear your feedback.
- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [x] If you changed the localization: Did you run `gradle localizationUpdate`?

## Notes

- [ ] BibSonomy seems to use following fields: `interhash;intrahash;keywords;groups;privnote`. Special handling for them is removed in this PR as it is unclear how they are used.

### CODE REVIEW

**Comment by tobiasdiez** [comment]
Does it make sense to integrate the "Search BibSononym" feature as a fetcher?

**Comment by koppor** [comment]
Here a screenshot:

grabbed_20160928-231813

**Comment by lenhard** [comment]
What is the status / plan here? I am not quite sure as to what questions are waiting for being answered here?

The size of the PR is a little scary. 5000+ lines are well above what we can meaningfully review.

**Comment by koppor** [comment]
@zellerdev is going to provide an update on Tuesday. The size of the code cannot be reduced easily, because we include a plugin, which is large by itself.

See https://github.com/JabRef/jabref/pull/2157/commits/f8da659de4b982d38ad4234056dd1a95241efe54 and https://bitbucket.org/bibsonomy/bibsonomy-jabref-plugin/wiki/Home

This PR is a huge step to fix https://github.com/JabRef/jabref/issues/152

**Comment by koppor** [comment]
Status: Functionality works. Also with [PUMA](https://puma.ub.uni-stuttgart.de/). Others should test it, too.

Open issues:

- [ ] Code refactoring
- [ ] Tests
- [ ] Check whether the handling of Metadata is correct

**Comment by koppor** [comment]
Any news? Timeline?

**Comment by koppor** [comment]
Currently, the preferences look as follows. I will clean them up:

grafik

File: src/main/java/org/jabref/gui/IconTheme.java [hunk 1]
```diff
@@ -48,16 +48,10 @@
     // Christmas edition
     //public static final Color DEFAULT_COLOR = new Color(0x155115);
     //public static final Color DEFAULT_DISABLED_COLOR = new Color(0x990000);
-    private static Font FONT_16;
```
Comment (koppor) [inline src/main/java/org/jabref/gui/IconTheme.java]: This should not have been removed as it came from the master branch. - I think, I got it back by a subsequent merge.

**Comment by lenhard** [comment]
Bump!

Any updates here? The longer this stays stale, the more it diverges from master.

**Comment by koppor** [comment]
The token is on my side, but priorities shifted. May in December as Xmas
present.

Am 27.10.2017 5:52 nachm. schrieb "Jörg Lenhard" <notifications@github.com>:

**Comment by koppor** [comment]
Blocked by https://bitbucket.org/Jerady/fontawesomefx/pull-requests/14/add-cancel-icon/diff.

**Comment by tobiasdiez** [comment]
Maybe @halirutan can add the `cancel` icon to our own font (or we just choose a different icon for the time being...)

**Comment by halirutan** [comment]
@koppor @tobiasdiez Here https://github.com/JabRef/jabref/pull/4069

**Comment by tobiasdiez** [comment]
@halirutan perfect! Thanks.
@koppor now you need to base on the maintable branch ;-)

**Comment by tobiasdiez** [comment]
Any update? I'll mark this as on-hold for the moment. Feel free to merge the latest master, update the code and reopen.

**Comment by koppor** [comment]
I am not able to code JavaFX, so we really need help from the community. Maybe @zesaro wants to learn JavaFX and bring JabRef a huge step forward!

**Comment by zesaro** [comment]
I am always interested in learning new things. Currently I'm not up to date with this PR. Is this meant to update this PR or address issues in the [Rework](https://github.com/JabRef/jabref/projects/2)?
But it eventually would be a slow process. What about the JavaFX Team from @JabRef/stupro  ? :smile:

**Comment by zesaro** [comment]
Is this still relevant? What has to be done here? Something with JavaFX?
I'd love to finish what I started 😅

**Comment by koppor** [comment]
Yes, definitvely! The UI has to be ported to JavaFX and the functionality
has to be checked. No popup if pseudo credentials are used. Only if
bibsonomy Dialog is shown.

Sascha Zeller <notifications@github.com> schrieb am Mi., 13. März 2019,
22:27:

**Comment by zesaro** [comment]
Is the bibsonomy-rest-client still usable with the current version of JabRef? I didn't work with Java for a long time now and never touched the new modules.
But I was curious. So I checked out a new branch from the current master and updated build.gradle to use the dependencies for the current bibsonomy-rest-client (3.9.2). Basically a two-liner.
But this seems to bite with some modules. At least I can't execute the run command successfully anymore.
Amy I missing something?

**Pull request closed without merging**